src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -960,34 +967,22 @@ private JsonRpcMessage HandleInvocationTaskResult(JToken id, Task t)
 
             if (t.IsFaulted)
             {
-                if (t.Exception != null)
+                if (this.IsFatalException(StripExceptionToInnerException(t.Exception)))
                 {
-                    if (t.Exception.InnerException != null)
-                    {
-                        if (this.IsFatalException(t.Exception.InnerException))
-                        {
-                            ExceptionDispatchInfo.Capture(t.Exception.InnerException).Throw();
-                        }
-                    }
-                    else
-                    {
-                        if (this.IsFatalException(t.Exception))
-                        {
-                            ExceptionDispatchInfo.Capture(t.Exception).Throw();
-                        }
-                    }
+                    var exception = StripExceptionToInnerException(t.Exception);
+                    var e = new JsonRpcDisconnectedEventArgs(
+                        string.Format(CultureInfo.CurrentCulture, Resources.FatalExceptionWasThrown, exception.GetType(), exception.Message),
+                        DisconnectedReason.Unknown,
```

Comment:
Why are we using `Unknown` as the reason when you've defined a new enum value specifically for this situation?